test/memoryProviderAPI.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// This file contains tests for UMF provider API
+
+#include "pool.h"
+#include "provider.h"
+#include "provider.hpp"
+
+#include <string>
+#include <unordered_map>
+
+using umf_test::test;
+
+TEST_F(test, memoryProviderTrace) {
+    static std::unordered_map<std::string, size_t> calls;
+    auto trace = [](const char *name) { calls[name]++; };
+
+    auto nullProvider = umf_test::wrapProviderUnique(nullProviderCreate());
```

Comment:
I'm not convinced about using wrapProviderUnique function instead of using regular UMF API (umfProviderCreate) here. I think it complicates the code and we also don't test here umfProviderCreate functions straightly.

Reply:
wrapProvider unique does not call any create function underneath, it only wraps the handle into unique_ptr - you still need to call umfProviderCreate (or nullProviderCreate as in this case) so we don't hide anything